### PR TITLE
Fix poll_read_buf signature

### PR DESCRIPTION
**Pull request by Darksonn**
Follow-up to: #3052

### CODE REVIEW

File: tokio/src/io/util/async_read_ext.rs [hunk 1]
```diff
@@ -250,9 +251,12 @@ cfg_io_util! {
         /// If this function encounters any form of I/O or other error, an error
         /// variant will be returned. If an error is returned then it must be
         /// guaranteed that no bytes were read.
-        /// ```
-        fn poll_read_buf<'a, B>(&'a mut self, buf: &'a mut B, cx: &mut Context<'_>) -> Poll<io::Result<usize>> where Self: Unpin + Sized, B: BufMut {
-            poll_read_buf(self, buf, cx)
+        fn poll_read_buf<B>(self: Pin<&mut Self>, cx: &mut Context<'_>, buf: &mut B) -> Poll<io::Result<usize>>
+        where
+            Self: Sized,
```
Comment (djc) [inline tokio/src/io/util/async_read_ext.rs:256]: Should this lose the `Sized` bound, then?

Comment (Darksonn) [reply tokio/src/io/util/async_read_ext.rs]: Thanks

**Review by taiki-e — APPROVED** [review]
Thanks for the fix! (and sorry for missing these in #3052's review.)

**Comment by Darksonn** [comment]
It's alright, I missed it too until after clicking merge :sweat_smile:

**Comment by carllerche** [comment]
Thanks for this :+1:

The PR should no longer be necessary as `poll_read_buf` is being pulled from `tokio`.

**Pull request closed without merging**